server/public/model/post.go

```diff
@@ -657,6 +664,152 @@ func (o *Post) GetProp(key string) any {
 	return o.Props[key]
 }
 
+// ValidateProps checks all known props for validity.
+// Currently, it logs warnings for invalid props rather than returning an error.
+// In a future version, this will be updated to return errors for invalid props.
+func (o *Post) ValidateProps(logger mlog.LoggerIFace) {
+	if err := o.propsIsValid(); err != nil {
+		logger.Warn(
+			"Invalid post props. In a future version this will result in an error. Please update your integration to be compliant.",
+			mlog.String("post_id", o.Id),
+			mlog.Err(err),
+		)
+	}
+
+	for i, a := range o.Attachments() {
+		if err := a.IsValid(); err != nil {
+			logger.Warn(
+				"Invalid messsage attachment. In a future version this will result in an error. Please update your integration to be compliant.",
+				mlog.String("post_id", o.Id),
+				mlog.Int("attachment_index", i),
+				mlog.Err(err))
+		}
+	}
```

Comment:
I think I understand the sentiment here, and it is not a bad approach, but I got a little confused. No need for changes, just raising the concern to verify if it is valid or not.

My confusion is because we have a "ValidateProps", that calls `propsIsValid`, and then validate the attachments. Since attachemnts is another prop, shouldn't it be inside `propsIsValid`?

As I said, this probably simplifies many other things, but I got a bit confused.

Reply:
Fair point. I've moved the attachment validation inside of `propsIsValid` via 0ee90254a42163e3f95ceb00ed82036c3358769d.